cpp/memilio/mobility/graph.h

```diff
@@ -178,6 +178,62 @@ class Graph
                                       });
     }
 
+    /**
+    * @brief Add edges to a graph without checking for duplicates and without sorting.
+    * 
+    * @param start_node_idx Id of start node
+    * @param end_node_idx Id of end node
+    * @param args Additional arguments for edge construction
+    * @return Edge<EdgePropertyT>& End of edge vector
+    */
+    template <class... Args>
+    Edge<EdgePropertyT>& lazy_add_edge(size_t start_node_idx, size_t end_node_idx, Args&&... args)
+    {
+        assert(m_nodes.size() > start_node_idx && m_nodes.size() > end_node_idx);
+        m_edges.emplace_back(start_node_idx, end_node_idx, std::forward<Args>(args)...);
+        return m_edges.back();
+    }
+
+    /**
+     * @brief Sort the edge vector of a graph.
+     * 
+     * @return Edge<EdgePropertyT>& End of edge vector
+     */
+    Edge<EdgePropertyT>& sort_edges()
+    {
+        std::sort(m_edges.begin(), m_edges.end(), [](auto&& e1, auto&& e2) {
+            return e1.start_node_idx == e2.start_node_idx ? e1.end_node_idx < e2.end_node_idx
+                                                          : e1.start_node_idx < e2.start_node_idx;
+        });
+        return m_edges.back();
+    }
+
+    /**
+     * @brief Make the edges of a graph unique.
+     * 
+     * Copies all the unique edges to a new vector and replaces the edge vector of the graph with it. Unique means that 
+     * the start and end node indices are unique. Other edge properties are not checked.
+     * @return Edge<EdgePropertyT>& End of edge vector
+     */
+    Edge<EdgePropertyT>& make_edges_unique()
```

Comment:
i think it would be clearer, if we use the term "remove dublicates" instead of unique edges. 
Also renaming the function to `remove_duplicate_edges` would add clarity here imo.